Add draft of 'how we built batch changes'

mrnugget opened the pull request:
This is a first draft meaning that I still want to polish it and rephrase certain things. ~Oh and also add the last two paragraphs. The idea and structure, though, are what I had in mind.~

Now that I've written this draft I'm not 100% sure whether this is interesting to readers of the Sourcegraph blog (on my personal blog I don't really have to care about such considerations).

Opinions? Thoughts?

ebrodymoore commented on blogposts/2021/how-we-built-batch-changes.md:
```diff
@@ -0,0 +1,195 @@
+
+Back to the five pull requests and how we've managed to defy both of these dangers.
+
+In September 2019 we started working on Batch Changes, except that back then it
```
```suggestion
In September 2019 after Eric Brody-Moore had the original idea we started working on Batch Changes, except that back then it
```

mrnugget replied:
```
In September 2019, _despite_ Eric Brody-Moore, we started working on Batch Changes, except that back then it
```

ebrodymoore replied:
Hmm it seems like you're trying to re-write history here 🤔

mrnugget replied:
That's _exactly_ what I'm trying to do, yes!

ebrodymoore commented on blogposts/2021/how-we-built-batch-changes.md:
```diff
@@ -0,0 +1,213 @@
+
+Only a few months later, though, we were stuck. We realised that what we had
+built and shipped as an alpha to customers just didn't work on a conceptual
+level. In that version, campaigns (now called batch changes) were executed on
```
The next question that could be answered here is `What went "wrong" (though it's not really wrong) in the product development that you built something that didn't work on a conceptual level?`

mrnugget replied:
Ah, yes. That _is_ interesting. I'm not sure I have an answer though (or if someone else could better write that), but I'll think about it.

mrnugget replied:
I can't up with something that I can confidently write about. I'll keep thinking about it though.

ebrodymoore commented on blogposts/2021/how-we-built-batch-changes.md:
```diff
@@ -0,0 +1,213 @@
+making our product as a whole more consistent. One thing that stood out was the
+name "Campaigns" that didn't fit next to the names of our other features and
+always required a sentence or two of explanation ("Glad you asked. A campaign is
+a ..."). The name Batch Changes, on the other hand, fared far better in surveys:
```
Maybe this isn't the time or place but could be cool to throw in a few sentences about the naming process. Or at least a couple other options we were considering!

mrnugget replied:
"Clusters" and "Bulk Changes" and "Batches" and "Campaigns" were other options and didn't do as well. I'll try to work that in!

A separate "Naming is hard, or: how to find a product name" blog post would be cool.

mrnugget replied:
Changed that paragraph

ebrodymoore reviewed (commented):
This is really awesome. So glad I get to be considered as a co-author on this. 

But in all seriousness, this was really cool for me to read and understand the different iterations you all went through because I definitely wasn't paying close enough attention to everything that has gone into Batch Changes. I left a couple questions that I now want answers to but it's a great post.

mrnugget commented on blogposts/2021/how-we-built-batch-changes.md:
```diff
@@ -0,0 +1,213 @@
+You also haven't built it (yet), so you don't have experience building it.
+Chances that you can accurately predict how its development will go are low.
+
+Acknowledging that you _don't_ plan the development of the product from start to
```
Ah, I see what you mean. Rephrased it and (hopefully) improved it.

chrispine commented:
This is great! I have lots of small editorial comments I could make, but I wasn't sure if this was at the right stage for those kinds of comments. Let me know when you want that sort of feedback.

eseliger commented on blogposts/2021/how-we-built-batch-changes.md:
```diff
@@ -0,0 +1,213 @@
+PR #1:
+- 5 Sep 2019: [a8n: Implement campaigns in GraphQL API by tsenart](https://github.com/sourcegraph/sourcegraph/pull/5482)
+PR #2:
+- 26 Jan 2021: [a8n: support creating campaign plans from API only, not UI by sqs](https://github.com/sourcegraph/sourcegraph/pull/8008)
+PR #3:
+- 20 Feb 2021: [rename (in docs and settings): automation -> campaigns by sqs](https://github.com/sourcegraph/sourcegraph/pull/8507)
```
these two years should probably be 2020?

mrnugget replied:
Yes! But these are just comments/notes that I used when collecting relevant PRs.

eseliger commented:
Idea: 
From an engineers perspective, I'd like to hear about how this process went and whether we're stuck with loads of legacy code now and for the benefit of a nice product and name now have to deal with a nasty codebase
There's some hints towards _no_, but maybe we can add a separate paragraph telling that it actually helped us _keep_ it clean, since we constantly reevaluate, throw out old stuff, and rethink existing stuff and don't stick to it "just because it's there and works". See reconciler/rewirer refactor and such, as part of throwing over the existing core system and migrating everything to be previewable

mrnugget commented:
> but maybe we can add a separate paragraph telling that it actually helped us _keep_ it clean

This is a really cool idea. I'll try to see if I can weave that in. (I can also see this as another post 🤔 "Build 2 to throw 2 away" in a way...)

mrnugget commented:
> I have lots of small editorial comments I could make, but I wasn't sure if this was at the right stage for those kinds of comments. Let me know when you want that sort of feedback.

Go for it!

chrispine commented on blogposts/2021/how-we-built-batch-changes.md:
```diff
@@ -0,0 +1,197 @@
+colleague to go from "ok, tell me what campaigns are" to "ohhh, I get it! nice"
+was reduced tenfold. Very much to our relief, since the changes necessary to
+make the documentation a reality were huge. We had to build a distributed,
+_declarative_ system that manages hundreds and thousands of pull and merge
```
```suggestion
_declarative_ system that manages hundreds or thousands of pull and merge
```
"and" isn't wrong here, but for some reason "or" sounds more idiomatic to me

chrispine commented on blogposts/2021/how-we-built-batch-changes.md:
```diff
@@ -0,0 +1,197 @@
+_declarative_ system that manages hundreds and thousands of pull and merge
+requests across different code hosts.
+
+Once the idea was validated by a lot "this totally makes sense" from potential
```
```suggestion
Once the idea was validated by a lot of "this totally makes sense" from potential
```

chrispine commented on blogposts/2021/how-we-built-batch-changes.md:
```diff
@@ -0,0 +1,197 @@
+requests across different code hosts.
+
+Once the idea was validated by a lot "this totally makes sense" from potential
+users we got to work. Look at [this PR][pr4b], number 4b in this series if you
```
```suggestion
users, we got to work. Look at [this PR][pr4b], number 4b in this series if you
```

chrispine commented on blogposts/2021/how-we-built-batch-changes.md:
```diff
@@ -0,0 +1,197 @@
+_iterating_. We added more features, fixed a lot of bugs, made things go faster.
+
+But — and by now you should know where this is going — something felt _off_.
+While Campaigns worked and we had customers using it and saying they loved it,
```
Just wanted to say that, in this case and the next several that follow, I like that you are using "Campaigns", because you are talking about renaming. So I wouldn't change it in these places. 🙂

chrispine commented on blogposts/2021/how-we-built-batch-changes.md:
```diff
@@ -0,0 +1,197 @@
+
+Towards the end of last year we ripped the "beta" label off of Campaigns and
+started to concentrate on getting more customers to use it: writing better (or
+even any) documentation, improving the onboarding process, providing
```
```suggestion
in some cases any) documentation, improving the onboarding process, providing
```

chrispine commented on blogposts/2021/how-we-built-batch-changes.md:
```diff
@@ -0,0 +1,197 @@
+Towards the end of last year we ripped the "beta" label off of Campaigns and
+started to concentrate on getting more customers to use it: writing better (or
+even any) documentation, improving the onboarding process, providing
+troubleshooting help, fixing bugs and edge cases.
```
```suggestion
troubleshooting help, and fixing bugs and edge cases.
```

chrispine commented on blogposts/2021/how-we-built-batch-changes.md:
```diff
@@ -0,0 +1,197 @@
+even any) documentation, improving the onboarding process, providing
+troubleshooting help, fixing bugs and edge cases.
+
+At the same time our fast-growing product marketing team started working towards
```
```suggestion
At the same time, our fast-growing product marketing team started working towards
```
This comma means there's only one way to parse this. Otherwise I have to read ahead to know how to parse it. Because maybe you were saying, "At the same time *that* our fast-growing..." in which case the sentence ends quite abruptly because the second half of it is missing.

chrispine commented on blogposts/2021/how-we-built-batch-changes.md:
```diff
@@ -0,0 +1,197 @@
+
+At the same time our fast-growing product marketing team started working towards
+making our product as a whole more consistent. One thing that stood out was the
+name "Campaigns". Next to the names of our other features it didn't fit in. It
```
```suggestion
name "Campaigns". Next to the names of our other features, it didn't fit in. It
```

chrispine commented on blogposts/2021/how-we-built-batch-changes.md:
```diff
@@ -0,0 +1,197 @@
+At the same time our fast-growing product marketing team started working towards
+making our product as a whole more consistent. One thing that stood out was the
+name "Campaigns". Next to the names of our other features it didn't fit in. It
+also always required a sentence or two of explanation ("Glad you asked. A
```
```suggestion
also always required a sentence or two of explanation. ("Glad you asked. A
```

chrispine commented on blogposts/2021/how-we-built-batch-changes.md:
```diff
@@ -0,0 +1,197 @@
+making our product as a whole more consistent. One thing that stood out was the
+name "Campaigns". Next to the names of our other features it didn't fit in. It
+also always required a sentence or two of explanation ("Glad you asked. A
+campaign is a ..."). In a data-driven approach to finding a new name Batch
```
```suggestion
campaign is a ...") In a data-driven approach to finding a new name, Batch
```

chrispine commented on blogposts/2021/how-we-built-batch-changes.md:
```diff
@@ -0,0 +1,197 @@
+also always required a sentence or two of explanation ("Glad you asked. A
+campaign is a ..."). In a data-driven approach to finding a new name Batch
+Changes came out as the winner. Ahead of Campaigns, Batches, Clusters, Bulk
+Changes, and others. No one asked in a survey needed an explanation and nearly
```
```suggestion
Changes, and others. No one we asked in our survey needed an explanation, and nearly
```

chrispine commented on blogposts/2021/how-we-built-batch-changes.md:
```diff
@@ -0,0 +1,197 @@
+campaign is a ..."). In a data-driven approach to finding a new name Batch
+Changes came out as the winner. Ahead of Campaigns, Batches, Clusters, Bulk
+Changes, and others. No one asked in a survey needed an explanation and nearly
+everyone understood what the feature is roughly about.
```
```suggestion
everyone understood what the feature was roughly about.
```

chrispine commented on blogposts/2021/how-we-built-batch-changes.md:
```diff
@@ -0,0 +1,197 @@
+But renaming what we built _again_? It would've been tempting to answer with
+"customers are already using it, why bother?" or "we've recorded demo videos
+with the old names!" or "there's so many screenshots we would need to change".
+In all honesty, though, we had to admit: existing customers probably won't mind
```
```suggestion
In all honesty, though, we had to admit: existing customers probably wouldn't mind
```

chrispine commented on blogposts/2021/how-we-built-batch-changes.md:
```diff
@@ -0,0 +1,197 @@
+non-alpha, non-beta, download-it-now-and-try-it launch of [Batch
+Changes][launch].
+
+Looking back over these five pull requests and at the past 1.5 years now I'm
```
```suggestion
Looking back over these five pull requests and at the past 1.5 years now, I'm
```

chrispine commented on blogposts/2021/how-we-built-batch-changes.md:
```diff
@@ -0,0 +1,197 @@
+
+Looking back over these five pull requests and at the past 1.5 years now I'm
+still surprised. I know how easy it is to say "but this is what we said we
+wanted to build" or "but we invested all that time!". I've worked in iterations
```
```suggestion
wanted to build" or "but we invested all that time!" I've worked in iterations
```

chrispine commented on blogposts/2021/how-we-built-batch-changes.md:
```diff
@@ -0,0 +1,197 @@
+still surprised. I know how easy it is to say "but this is what we said we
+wanted to build" or "but we invested all that time!". I've worked in iterations
+all my life. 2 week sprints, 4 weeks, Agile, Scrum, even - ugh - [SAFe][safe].
+The danger to get stuck and not knowing how to get out of it is always there.
```
```suggestion
The danger of getting stuck and not knowing how to get out of it is always there.
```

chrispine commented on blogposts/2021/how-we-built-batch-changes.md:
```diff
@@ -0,0 +1,197 @@
+I'm still not quite sure how we avoided that trap. My best guess (and this
+_will_ sound cheesy) is: commitment. Commitment to building an excellent
+product, commitment to building something that provides value to customers, even
+if it means starting from scratch when you realise you're in a dead end.
```
```suggestion
if it means starting from scratch when you realize you're at a dead end.
```

chrispine reviewed (approved):
OK!! Lots of suggestions, and you should consider ignoring at least some of them. I am confident that most of my suggestions sound clearer/better to speakers of _American_ English, but your mileage (kilometerage?) may vary. 😄 

If Adam reviews this after I do, he'll change "realize" back to "realise", and probably suggest other changes, and those will also be correct. 🤷🏻‍♀️

mrnugget commented on blogposts/2021/how-we-built-batch-changes.md:
```diff
@@ -0,0 +1,197 @@
+Exciting, rewarding, surprising, iterative, customer-focused - one of those
+maybe, but not straightfoward. The two name changes in 1.5 years of development
+— from Automation to Campaigns to Batch Changes — give you a glimpse of why. The
+real story, though, the one that makes me doubt whether we actually pulled this
```
Good point! I rephrased this.

mrnugget commented on blogposts/2021/how-we-built-batch-changes.md:
```diff
@@ -0,0 +1,197 @@
+that building the product from the ground up, fully understanding and owning the
+code is more important to the long-term success of the project and the team than
+merging a prototype to get something out there as fast as possible. From my
+perspective now this was one of the key decisions that, to this day, helped us
```
Thanks!

mrnugget commented on blogposts/2021/how-we-built-batch-changes.md:
```diff
@@ -0,0 +1,197 @@
+level. In that version, campaigns (now called batch changes) were executed on
+the Sourcegraph instance to produce changes in repositories, which was cool from
+a technical standpoint but it felt clunky, and slow, and adding a new type of
+campaign (to produce a new type of change) required us to add new code. "Meh" is
```
Rephrased the sentence to avoid the name.

mrnugget commented on blogposts/2021/how-we-built-batch-changes.md:
```diff
@@ -0,0 +1,197 @@
+way to ship successful products.
+
+But there's one huge problem with this approach: it's really easy to iterate
+yourself into a corner. If all you do is take small steps, you can miss the
```
This kind of feedback is _my jam_ as they say (or ... said? I don't know)

I changed this paragraph to make the metaphor more consistent, please take a look.

(I also had more ideas and actually wanted to bring "dance" into this — "While you're happily doing this two-steps-forward-two-steps-back dance, you might miss when ..." — but let's see if this works too. I'm _always_ happy to go all in on metaphors.)

chrispine replied:
I left one small suggestion, but otherwise it looks good!

chrispine commented on blogposts/2021/how-we-built-batch-changes.md:
```diff
@@ -0,0 +1,201 @@
+
+But there's one huge problem with this approach: if you're constantly taking
+small steps and adjusting the direction slightly, you _feel_ productive even
+when you're going in circles. You're constantly doing two steps forward and two
```
```suggestion
when you're going in circles. You keep taking two steps forward and two
```

content-hacker commented on blogposts/2021/how-we-built-batch-changes.md:
```diff
@@ -0,0 +1,202 @@
+abstraction layers it introduced turned out to be incredibly well-fitting. They
+allowed us to go faster and get a _valuable_ tool into the hands of our
+customers even earlier.
+
```
```suggestion
## Pull requests 3 & 4: Still not there... more fine-tuning and user testing
```

mrnugget replied:
I used two headings for #3 and #4 (and also realised that PR#4 had the wrong URL)

content-hacker commented on blogposts/2021/how-we-built-batch-changes.md:
```diff
@@ -0,0 +1,202 @@
+for the curious: the code today is [even cooler][evencoolercode], take a look).
+
+That brings us to the last of the five pull requests.
+
```
```suggestion
## Pull request 5: Main point of this PR reference
```

mrnugget replied:
I have to admit that it took me really long to understand that "Main point of this PR reference" was directed at me 😄

content-hacker reviewed (changes requested):
@mrnugget this is a great account of how batch changes were built! You've captured great lessons learned and exemplified thought-provoking problem-solving.  I think our users will appreciate knowing the care that went into building this, and other developers can learn from this experience. 

My review focuses on the structure of the piece and suggested edits for overall clarity and readability. I've captured the high-level notes here. 

1. The structure of the post is hard to follow in the beginning. You oscillate between talking about iterative development and your personal experience with it. My suggestion is to open more generally, focusing on iterative development to set up the scene and appeal to your audience. 

2. The current post leads with “I” and “me” statements. Consider leading with iteration-based work and a common pain point to build trust and recognition with your audience before asking them to follow along with your personal journey. 

3. Suggested structure:

- Introduce your topic: iteration-based development
- Speak to a common pain point: iterative development is a very rewarding way to work but it can feel counterintuitive; there are pitfalls. 
- Transition to your personal experience with iterative development 
- Subdivide your article contents according to each pull request. 

4. This is a long format post. Add section titles to improve readability, and to help the reader grasp the most important takeaways.

content-hacker commented on blogposts/2021/how-we-built-batch-changes.md:
```diff
@@ -0,0 +1,202 @@
+simply delete it! Right?" Right, says the [sunken cost fallacy][sunkencost].
+
+Back to the five pull requests and how we've managed to defy both of these dangers.
+
```
It would be nice to capture screen shots of the pull requests if they can help the reader better understand the story.

mrnugget replied:
Good idea! I uploaded screenshots of the PRs and added them at the top of each section.

content-hacker commented on blogposts/2021/how-we-built-batch-changes.md:
```diff
@@ -0,0 +1,202 @@
+---
+title: 'How we built Batch Changes, explained in 5 pull requests'
```
or just "How we avoided the biggest dangers of iteration-base product development"

mrnugget replied:
I really think we should keep the "5 pull requests" in there somehow, because that's what would make me click. "iteration-based product development" in a header is something I would skip. But "you build something in 5 PRs? which ones?" is what would make me click & read.

chrispine replied:
Agreed, I was immediately curious: a story told in 5 PRs? Which ones?

content-hacker replied:
What about: 

"How we avoided the biggest dangers of iteration-based product development explained in 5 pull requests"

content-hacker replied:
Shorter version: "Avoiding the pitfalls of iteration-based development explained in 5 pull requests"

mrnugget replied:
Hm, I think it depends on the audience we're targeting here. "How we built X" is always interesting to me, especially on a tech company blog. Stuff that sounds "product development-y" is hit and miss to me.

My original idea was:

```
How we built Batch Changes - in 5 pull requests
```

content-hacker replied:
My concern with leading with Batch Changes is that many people have never heard of it. Our audience for this post is software engineers, but not necessarily currents users, or even someone who has heard of Sourcegraph before. 

My question for you: is "how we built X" is universally interesting, regardless of product and company familiarity?

mrnugget replied:
> My question for you: is "how we built X" is universally interesting, regardless of product and company familiarity?

I wouldn't say regardless of company familiarity. When I know that the company builds interesting stuff I will definitely be more likely to read.

But given two options:

1. "How we built X" with X being a thing that sounds like a product name I don't know about
2. more generic sounding title

and the same company familiarity, I'd still prefer 1 more.

malomarrec replied:
I love the "in 5 pull requests" idea. This immediately hooks me to the article. Wait. What? Oh, nice. What I like from a more analytical perspective is that this title doesn't sound like other titles I'm used too. And I want to know why!! So I keep on reading.

mrnugget commented on blogposts/2021/how-we-built-batch-changes.md:
```diff
@@ -0,0 +1,202 @@
+show how we avoided two big dangers of iteration-based product development:
+local maximums and the sunken cost fallacy.
+
+To give you some context for these terms, let me tell you how I understand the
+premise of iteration-based work. It doesn't matter whether you call it _Agile_,
+_Sprint_, _Shape Up_ or something else, the idea is this: until you've built the
+product, you don't _really_ know what it should look like or how to best build
+it. You've never used it — how would you know how it works best? You also
+haven't built it yet, which means you don't have experience building it. Chances
+are low that you can accurately predict how its development will go.
+
+So what you do is you acknowledge that you don't know. And then you lean into that:
+you _don't_ plan the development of the product from start to finish in every
+detail. You _don't_ nail down and define the hundreds of steps it will take to
+go from zero to fully-built and shipped and launched product. Instead, working
+in iterations, you define roughly where you want to go and then take a single
+small step. Then you look around, get your bearings, correct course if
+necessary, take another step. Build something, ship it, learn, build again.
```
> Speaking as the author of the post you don't need to add qualifiers.

I still would like to leave them in, since I don't think it's a universal explanation for the concept (others _have_ different definitions) but a rather subjective view.

mrnugget replied:
I moved the paragraphs up, incorporated some wording suggestions, but left others in. Let me know if that works.

mrnugget commented:
> * The structure of the post is hard to follow in the beginning. You oscillate between talking about iterative development and your personal experience with it. My suggestion is to open more generally, focusing on iterative development to set up the scene and appeal to your audience.
> * The current post leads with “I” and “me” statements. Consider leading with iteration-based work and a common pain point to build trust and recognition with your audience before asking them to follow along with your personal journey.

I kinda disagree with these two points. And I think that maybe this is due to us having different ideas here of what this post could/should be.

I do like writing from a subjective point of view, because that's what I know. I can't give the reader a general, broad definition of iteration-based development. That's not my expertise and it's also not what I would do in conversation or when giving a talk. 

That's also what I want I want this post to be: personal story and point of view of the development of a product.

Leading with a generic definition and then making it concrete by giving examples reminds me of a lot of other company blog posts that are often a bit bland: "How we used test-driven development to increase our development velocity", "How continuous integration helped us recover from technical debt" etc. While the content can often contain interesting insights, when reading one of these I usually skip the fluff, because I want to get to the unique pieces: I don't want another definition of "continous intergration" or "test-driven development" if it's not super unique and contains insight. And if it is unique then why not zoom in on that and leave out the rest?

So for me personall, the opposite of this is true: 

> * The current post leads with “I” and “me” statements. Consider leading with iteration-based work and a common pain point to build trust and recognition with your audience before asking them to follow along with your personal journey.

When I read something that starts subjective I know that I usually get something new, something that I can't find on wikipedia.

And in this case, I think coming right out and saying "hey, this was _not_ a straightforward process. it might even look like the opposite to you - we've had 2 renames in 1.5 years" would build _more_ trust than saying "here's how we avoided X"

content-hacker commented:
> I do like writing from a subjective point of view, because that's what I know. I can't give the reader a general, broad definition of iteration-based development. That's not my expertise and it's also not what I would do in conversation or when giving a talk.

I'm not suggesting that you write from an objective or generic point of view. This post is a personal essay about what you experienced and thus is entirely subjective by nature. My suggestion is to pull the interesting insights from your experience building Batch Changes to the forefront of the article. My role as an editor is to help you get the best possible draft and optimize it for maximum views, specifically trying to broaden its appeal to people who have never heard of the product or company. 

When I suggest lead-in with iterative development (and this was just a suggestion, maybe there is a better topic to lead with), the intention is to give the reader a topic to orient themselves to before diving into your personal experience. I do strongly believe that the title should be elevated to a subject that is larger than Batch Changes, and agree that including "5 pull requests" is interesting. 

Ultimately, it is your article, and I would never try to force you to publish something you are not comfortable with. In my review, I made suggestions to make the piece clearer for readers. You can take what works for you, and leave what doesn't. I do encourage you to review our [content guidelines](https://about-docsite.sourcegraph.com/handbook/marketing/content#content-principles) and [writing for clarity checklist](https://docs.google.com/document/d/1KHFMIGmbmDPsE8EYPAf4DPD6kek_1dRnKf49wG5foU0/edit#heading=h.7ah9gwolqa2s), which is what I reference when doing any review. 

Let me know when your final draft is ready and I will do a copyedit and get it published. If you are curious, you can also check out the animation the brand team put together for sharing on social: https://app.asana.com/0/1199711242382929/1199989038479807/f

mrnugget commented on blogposts/2021/how-we-built-batch-changes.md:
```diff
@@ -0,0 +1,202 @@
+small step. Then you look around, get your bearings, correct course if
+necessary, take another step. Build something, ship it, learn, build again.
+
+I personally love working like this and think that in a lot of contexts in which
```
Done!

mrnugget commented on blogposts/2021/how-we-built-batch-changes.md:
```diff
@@ -0,0 +1,202 @@
+But there's one huge problem with this approach: if you're constantly taking
+small steps and adjusting the direction slightly, you _feel_ productive even
+when you're going in circles. You're constantly doing two steps forward and two
+steps back but never take a big leap to someplace else entirely. You end up with
```
Done!

mrnugget commented on blogposts/2021/how-we-built-batch-changes.md:
```diff
@@ -0,0 +1,202 @@
+small steps and adjusting the direction slightly, you _feel_ productive even
+when you're going in circles. You're constantly doing two steps forward and two
+steps back but never take a big leap to someplace else entirely. You end up with
+[a local maximum][localmaximum]. And even if you realise that you're stuck and
```
I added a short explanation, but kept the link. Let me know if it makes sense 😬

mrnugget commented on blogposts/2021/how-we-built-batch-changes.md:
```diff
@@ -0,0 +1,202 @@
+[a local maximum][localmaximum]. And even if you realise that you're stuck and
+_know_ that you must take a big leap, it's hard to abandon the things you've
+built in many, many small iterations. "So much work went into this! We can't
+simply delete it! Right?" Right, says the [sunken cost fallacy][sunkencost].
```
Added a sentence that I hope is clear - let me know if that's not the case

mrnugget commented on blogposts/2021/how-we-built-batch-changes.md:
```diff
@@ -0,0 +1,202 @@
+---
+title: 'How we built Batch Changes, explained in 5 pull requests'
+description: A 300 character limit field for describing your post. Use this is you want to specially craft the excerpt shown on the index page. Uses the first 300 characters of text from your post if this field does not exist.
```
Done!

mrnugget commented on blogposts/2021/how-we-built-batch-changes.md:
```diff
@@ -0,0 +1,202 @@
+_know_ that you must take a big leap, it's hard to abandon the things you've
+built in many, many small iterations. "So much work went into this! We can't
+simply delete it! Right?" Right, says the [sunken cost fallacy][sunkencost].
+
```
Done!

mrnugget commented on blogposts/2021/how-we-built-batch-changes.md:
```diff
@@ -0,0 +1,202 @@
+[a local maximum][localmaximum]. And even if you realise that you're stuck and
+_know_ that you must take a big leap, it's hard to abandon the things you've
+built in many, many small iterations. "So much work went into this! We can't
+simply delete it! Right?" Right, says the [sunken cost fallacy][sunkencost].
```
I kinda rephrased this a bit before I saw this exact comment, but I still think it works in the new structure - let me know if that's not the case.

mrnugget commented on blogposts/2021/how-we-built-batch-changes.md:
```diff
@@ -0,0 +1,202 @@
+published: false
+---
+
+"Straightforward" is the last word I'd use to describe how we built Batch
```
Done!

mrnugget commented:
>  My role as an editor is to help you get the best possible draft and optimize it for maximum views, specifically trying to broaden its appeal to people who have never heard of the product or company. [...]
> Ultimately, it is your article, and I would never try to force you to publish something you are not comfortable with. In my review, I made suggestions to make the piece clearer for readers. You can take what works for you, and leave what doesn't. I do encourage you to review our [content guidelines](https://about-docsite.sourcegraph.com/handbook/marketing/content#content-principles) and [writing for clarity checklist](https://docs.google.com/document/d/1KHFMIGmbmDPsE8EYPAf4DPD6kek_1dRnKf49wG5foU0/edit#heading=h.7ah9gwolqa2s), which is what I reference when doing any review.

Sorry, I think I got too defensive when seeing the "content-hacker requested changes" here and couldn't see yet (through the multiple GitHub comments) what you were actually trying to do here. I mistook "here's some well-meant suggestions" as "this is wrong, you should do this instead" - error on my part and probably due to my writing ego not being used to reviews as much as my coding ego.

In any case: I now incorporated most of your suggestions and am glad I did. (I also renamed the file, which means that the old comments disappeared here in GitHub, which might make reviewing this harder than it could be - sorry, but I think the file name should match the blog post title/slug)

So, I think this is now my final draft - please take another look and let me know if there's more to change.

---

> Let me know when your final draft is ready and I will do a copyedit and get it published. If you are curious, you can also check out the animation the brand team put together for sharing on social: https://app.asana.com/0/1199711242382929/1199989038479807/f

I don't have access to Asana 😬 I can get access, I think, but maybe (and I don't know, I've never seen Asana from the inside) it's easier if you drop the picture in here?

That reminds me: **social image and hero image are still missing**. I'm personally not a huge fan of hero images in blog posts, so if this one doesn't have one that's fine with me. But I don't have a really strong opinion here, so feel free to change.

content-hacker commented:
> Sorry, I think I got too defensive when seeing the "content-hacker requested changes" here and couldn't see yet (through the multiple GitHub comments) what you were actually trying to do here. I mistook "here's some well-meant suggestions" as "this is wrong, you should do this instead" - error on my part and probably due to my writing ego not being used to reviews as much as my coding ego.

I totally get that, and to be honest, reviewing and giving feedback on blog content via a GitHub pull request is less than optimal (for me, at least). I'm personally much more used to reviewing in a google doc. Would it have been better if I marked `leave review comments` versus `request changes`? It's quite possible that these options have connotations I am not aware of; I didn't have these options when I worked in GitLab. 

> So, I think this is now my final draft - please take another look and let me know if there's more to change.

This looks great! I just completed my copyedit (checking grammar/punctuation) and will publish it this afternoon (UTC-6). I'm going to schedule the social post for tomorrow (2021-04-16) so I can set it to publish at a high-traffic time on Twitter. 

> That reminds me: **social image and hero image are still missing**. I'm personally not a huge fan of hero images in blog posts, so if this one doesn't have one that's fine with me. But I don't have a really strong opinion here, so feel free to change.

I agree we don't need to add a hero image and removed this from the frontmatter. The GIF the brand team created should be enough to draw attention to the post on social. 

Thanks for your hard work on this!
